src/ray/raylet/lineage_cache.cc

```diff
@@ -286,12 +295,22 @@ void PopAncestorTasks(const UniqueID &task_id, Lineage &lineage) {
 }
 
 void LineageCache::HandleEntryCommitted(const UniqueID &task_id) {
+  RAY_LOG(DEBUG) << "task committed: " << task_id;
```

Comment:
Under the current design, we can't actually guarantee that this method won't fire twice (or more) for the same task ID, right? Couldn't that cause issues here?

Reply:
Hmm, it should be okay if this method fires twice. I think this function is idempotent.

Reply:
Also, the only case I can think of right now where this would fire twice is if a task gets reconstructed and added to the table again. That should be pretty rare.

Reply:
You were right :) I changed this to check that the status is already `COMMITTED` if the call to set the status fails.